src/Illuminate/Support/ServiceProvider.php

```diff
@@ -72,6 +73,33 @@ protected function loadRoutesFrom($path)
         }
     }
 
+    /**
+     * Recursively scan the directory at the given path
+     * and load any route files found if not already cached.
+     *
+     * @param  string  $path
+     * @return void
+     */
+    protected function loadAllRoutesFrom($path)
```

Comment:
Didn't you mean to add these to the existing `RouteServiceProvider` instead?, I dont think wvery SeviceProvider needs to have the ability to load routes.